x-pack/plugins/alerting/server/rules_client/rules_client.ts

```diff
@@ -2142,8 +2146,18 @@ export class RulesClient {
     // If snoozeEndTime is -1, instead mute all
     const newAttrs =
       snoozeEndTime === -1
-        ? { muteAll: true, snoozeEndTime: null }
-        : { snoozeEndTime: new Date(snoozeEndTime).toISOString(), muteAll: false };
+        ? {
+            muteAll: true,
+            snoozeSchedule: clearUnscheduledSnooze(attributes),
+          }
+        : {
+            snoozeSchedule: (attributes.snoozeSchedule ?? []).concat({
```

Comment:
I think we'll want to do the `clearUnscheduledSnooze()` here as well?  If the user creates a duration-based snooze, and then creates another one, won't there then be two?  I guess that raises the question, is that a valid thing to do - have multiple duration-based snoozes without a real rRule schedule?  Seems like we should only have one, and it would be the last one the user set.  For example:

- time X, set snooze for 1 hour
- time X+1m, set snooze for 10 minutes

How long do we snooze?  X + 1h or X + 11m (X+1m + 10m snooze)?